crates/format/src/corpus.rs

```diff
@@ -35,33 +35,47 @@ impl Corpus {
 ///
 /// `path` is expected to be a directory.
 pub fn collect_metadata(path: &Path, tests: &mut Vec<MetadataFile>) {
-    let dir_entry = match std::fs::read_dir(path) {
-        Ok(dir_entry) => dir_entry,
-        Err(error) => {
-            tracing::error!("failed to read dir '{}': {error}", path.display());
-            return;
-        }
-    };
-
-    for entry in dir_entry {
-        let entry = match entry {
-            Ok(entry) => entry,
+    if path.is_dir() {
+        let dir_entry = match std::fs::read_dir(path) {
+            Ok(dir_entry) => dir_entry,
             Err(error) => {
-                tracing::error!("error reading dir entry: {error}");
-                continue;
+                tracing::error!("failed to read dir '{}': {error}", path.display());
+                return;
             }
         };
 
-        let path = entry.path();
-        if path.is_dir() {
-            collect_metadata(&path, tests);
-            continue;
-        }
+        for entry in dir_entry {
+            let entry = match entry {
+                Ok(entry) => entry,
+                Err(error) => {
+                    tracing::error!("error reading dir entry: {error}");
+                    continue;
+                }
+            };
+
+            let path = entry.path();
+            if path.is_dir() {
+                collect_metadata(&path, tests);
+                continue;
```

Comment:
```suggestion
        collect_metadata(&path, tests);
```

When `collect_metadata` is called, we now check whether the path is a dir or file anyway, so no need to also check it here :)